src/decode.rs

```diff
@@ -46,6 +47,17 @@ impl fmt::Display for Error {
 #[cfg(feature = "std")]
 impl std::error::Error for Error {}
 
+/// Only available when the feature `std` is present.
+#[cfg(feature = "std")]
+impl Into<std::io::Error> for Error {
+    fn into(self) -> std::io::Error {
+        match self {
+            Error::Insufficient => std::io::ErrorKind::UnexpectedEof,
+            Error::Overflow => std::io::ErrorKind::InvalidData,
+        }.into()
```

Comment:
```suggestion
        let kind = match self {
            Error::Insufficient => std::io::ErrorKind::UnexpectedEof,
            Error::Overflow => std::io::ErrorKind::InvalidData,
        };
        std::io::Error::new(kind, self)
```

Any particular reason why not include the original error?

Reply:
It just slipped my mind.